### PR TITLE
Cumulative patch for select issues

### PR DESCRIPTION
**Pull request by lfryc**
I have migrated following patches from RichFaces 5 to RichFaces 4.5:

https://issues.jboss.org/browse/RF-11021
https://issues.jboss.org/browse/RF-11088
https://issues.jboss.org/browse/RF-12613
https://issues.jboss.org/browse/RF-13101

### CODE REVIEW

**Comment by jigg4** [comment]
Hello @lfryc , i tried to build the demo-examples, but received the following compilation error:
dev-examples-4.5.x\input-demo\src\main\java\org\richfaces\demo\RichFileUploadBean.java:[24,30] error: cannot find symbol.

I have no clue why this is happening, because there were absolutely no changes to the mentioned file as far as i know.

**Comment by lfryc** [comment]
Hey @jigg4 , works for me:

```
git clone https://github.com/richfaces4/components.git
cd components
git checkout cumulative-patch-for-select-issues
mvn install
cd ..

git clone https://github.com/richfaces4/dev-examples.git
cd dev-examples
git checkout 4.5.x
mvn install
```

```
$ mvn -v
Apache Maven 3.0.5 (r01de14724cdef164cd33c7c8c2fe155faf9602da; 2013-02-19 14:51:28+0100)
Maven home: ~/Apps/apache-maven-3.0.5
Java version: 1.7.0_19, vendor: Oracle Corporation
Java home: /usr/lib/jvm/java-1.7.0-openjdk-1.7.0.19.x86_64/jre
Default locale: en_US, platform encoding: UTF-8
OS name: "linux", version: "3.9.4-200.fc18.x86_64", arch: "amd64", family: "unix"
```

**Comment by jigg4** [comment]
Hello @lfryc ,
i still have the same problem. 
can the Picklist Probleme could cause this issue? because i can build components only if i delete the not supported symbols in components\input\ui\src\main\java\org\richfaces\component\AbstractPickList.java (add, add all,delete...), while i do receive an incorrect error after the components build fails: it is referring to faces-config from inpout-ui, but in the faces-config everything seems fine.
I am using windows btw.

**Comment by lfryc** [comment]
We need to verify the issue in Metamer [1] as `dev-examples:4.5.x` are not stable.

[1] https://issues.jboss.org/browse/RFPL-3240

**Comment by manovotn** [comment]
I tried to verify this with Metamer, adding one extra rich:select mentioned in RF-12613. Using this patch rich:select does not work at all. I am unable to pick any option at all as long as enableManualInput=true.

Setting this attribute to false I have following results:

RF-11021 - first item should be shown when there is no default label
- works with both selects on page

RF-11088 - when item is selected, there should be all the option in option list
- works with both selects

RF-12613 - selecting wrong items
- cannot verify since this was reported with manualInput = true (however picking "test" results in choosing the correct item now, which was previously problem ever without manual input)

EDIT: added information about following issue
RF-13101 - method getSelectInputLabel
- cannot reproduce bug on any 4.5 version (and not even on 4.3.x), I am able to use Object in itemValue as well as in f:selectItems (eg. in List<SelectItem> passes to it)
- please let me know if this bug was meant differently

**Comment by lfryc** [comment]
@manovotn  thanks for verification - so let me re-phrase so that I can make sure I understand it correctly:

RF-11021, RF-11088 clearly fixes the issue = you was able to reproduce issue and verify that with patch they no longer occurs

RF-12613 not sure I understand why you can't verify it with `enableManualInput=true`

RF-13101, let's see whether @jigg4  could provide a reproducer here, I admit I can't understand the description of this issue clearly as well

**Comment by manovotn** [comment]
@lfryc no, the patch only fixes some of the problems when `enableManualInput=false`, if you set it to true the component does not work at all! You are unable to select any option. I said this in the first few lines, then I tried the same while disabling manual input.

Sorry I might have made a mess of it because I tried to track down whether this patch works with `enableManualInput=false` (realizing there might be a bug related to this since the issues were focused around this attribute, e.g. RF-12613 was reported for manualInput set to true). So the rest of the results in my previous comment are with the manualInput set to false (I said that in my previous comment but I admit I should have made it bold or something, I am not quite familiar with GitHub Markdown in comments yet).

So to sum it up - as it is, the component is not working at all with `enableManualInput=true`

**Comment by lfryc** [comment]
Thanks for clarification, @manovotn!

**Comment by lfryc** [comment]
It's possible that the problems with `enableManualInput=false` are related to the fact we have applied all the patches at all.

@jigg4 do the patches work for you separately?

**Comment by lfryc** [comment]
I'm afraid I need to reject those patches as they destabilize the component.

**Comment by lfryc** [comment]
@jigg4 let me please know whether you would like to work on the patch/patches and incorporate them separately/as a whole into RichFaces.

**Comment by jigg4** [comment]
@lfryc i definetly would like to review the patches and incorporate them seperetly.
